telos-frontend/src/components/todo/DashboardTodo.jsx

```diff
@@ -1,3 +1,376 @@
+      due: '2021-04-30',
+      onGoing: true,
+      completed: false,
+    },
+    {
+      name: 'OutDated',
+      due: '2021-01-30',
+      onGoing: false,
+      completed: true,
+    },
+    {
+      name: '一二三四五',
+      due: '2021-04-30',
+      onGoing: true,
+      completed: true,
+    },
+    {
+      name: '上山打老虎',
+      due: '2021-04-30',
+      onGoing: true,
+      completed: false,
+    },
+  ];
+
+  const classes = useStyles();
+  const [item, setItem] = useState('');
+  const [dueDate, setDueDate] = useState('2021-01-01');
+  const [itemList, setItemList] = useState(listitems);
+  const [newItem, setNewItem] = useState(itemList);
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [cancel, setCancel] = useState([0]);
+  const [reDate, setReDate] = useState('');
+  const [reEventName, setReEventName] = useState('');
+  const [reItem, setReItem] = useState('');
+
+  const reDateChange = (event) => {
+    setReDate(event.target.value);
+    setReItem({ name: reItem.name, due: event.target.value, onGoing: true, completed: false });
+  };
+
+  const handleToggle = (value) => () => {
+    let newList = [...newItem];
+    for (let x of newList) {
+      if (x.name === value.name) {
+        x.completed = !x.completed;
+      }
+    }
+
+    setNewItem(newList);
+    setItemList(newList);
+  };
+
+  const firstEvent = (event) => {
+    setItem({ name: event.target.value, due: dueDate, onGoing: true, completed: false });
+  };
+
+  const dateChange = (event) => {
+    setDueDate(event.target.value);
+    setItem({ name: item.name, due: event.target.value, onGoing: true, completed: false });
+  };
+
+  const secondEvent = () => {
+    setNewItem((prev) => [...prev, item]);
+    setItem('');
+  };
+
+  const handleOption = (value) => (event) => {
+    setReEventName(value);
+    // reEventName = value;
+    setReItem({ name: value, due: reDate, onGoing: true, completed: false });
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleOptionClose = (event) => {
+    setAnchorEl(null);
+  };
+
+  const inputAlert = () => {
+    alert('Please enter something :)');
+  };
+
+  const cancelevent = (value) => () => {
+    const currentIndex = cancel.indexOf(value);
+    const newCancel = [...cancel];
+
+    if (currentIndex === -1) {
+      newCancel.push(value);
+    } else {
+      newCancel.splice(currentIndex, 1);
+    }
+    setCancel(newCancel);
+  };
+
+  const deleteevent = (value) => () => {
+    let newList = [...newItem];
+    for (let x of newList) {
+      if (x.name === value) {
+        const index = newList.indexOf(x);
+        newList.splice(index, 1);
+      }
+    }
+
+    setNewItem(newList);
+    setItemList(newList);
+  };
+
+  const scheduleevent = () => {
+    let newList = [...newItem];
+    let reItemx = reItem;
+
+    for (let x of newList) {
+      if (x.name === reItemx.name) {
+        x.due = reItemx.due;
+      }
+    }
+
+    setNewItem(newList);
+    setItemList(newList);
+  };
+
+  return (
```

Comment:
This does not follow the design defined [in the wiki](https://github.com/UOA-SE701-Group3-2021/3Lancers/wiki/To-Do-List#hi-fi-design-2). If you wish to change the design, you should open a new issue and wait for it to be approved. Changing the design is not within the scope of this issue.